src/server/actions/coder.actions.ts

```diff
@@ -0,0 +1,18 @@
+'use server'
+
+import { createUserAndWorkspace } from '../coder'
+import { Action, z } from './action'
+
+export const createUserAndWorkspaceAction = new Action('createUserAndWorkspaceAction', { performsMutations: true })
+    .params(
+        z.object({
+            name: z.string().nonempty(),
+            studyId: z.string().nonempty(),
+        }),
+    )
+    .handler(async ({ params: { name, studyId }, session }) => {
+        if (!session) throw new Error('Unauthorized')
```

Comment:
add a `requireAbilityTo('create', 'study')` and then session will be guaranteed to be defined.